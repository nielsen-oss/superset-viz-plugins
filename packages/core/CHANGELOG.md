# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.1.1 (2020-12-30)

**Note:** Version bump only for package @superset-viz-plugins/core





# 0.1.0 (2020-12-28)


### Features

* basic core package lib ([5f30462](https://github.com/nielsen-oss/superset-viz-plugins/commit/5f304621fbeea6554740eb2bfd156338f772429e))
