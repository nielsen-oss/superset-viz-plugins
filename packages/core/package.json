{
  "name": "@superset-viz-plugins/core",
  "version": "0.1.1",
  "description": "Superset-viz-plugins UI core package",
  "main": "lib/index.js",
  "module": "esm/index.js",
  "files": [
    "esm",
    "lib"
  ],
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.com"
  },
  "scripts": {
    "test": "jest --coverage --verbose --passWithNoTests",
    "tdd": "NODE_ENV=test jest --watch",
    "build": "NODE_ENV=development webpack --mode=development --colors",
    "build:watch": "NODE_ENV=development webpack --mode=development --colors --watch",
    "build:prod": "BABEL_ENV=production babel src -d dist --verbose --extensions \".ts,.tsx,.js,.jsx \" --ignore \"src/**/*.test.tsx\",\"src/*.test.js\",\"src/**/*.mock.js\",\"src/**/*.stories.tsx\",\"src/**/__stories__/*\",\"src/**/__mocks__/*\",\"src/**/__tests__/*\"",
    "build:esm": "ESM=true BABEL_ENV=production babel src -d esm --verbose --extensions \".ts,.tsx,.js,.jsx \" --ignore \"src/**/*.test.tsx\",\"src/*.test.js\",\"src/**/*.mock.js\",\"src/**/*.stories.tsx\",\"src/**/__stories__/*\",\"src/**/__mocks__/*\",\"src/**/__tests__/*\""
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com:nielsen-oss/superset-viz-plugins.git"
  },
  "keywords": [
    "superset",
    "visualization",
    "analysis",
    "data"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/nielsen-oss/superset-viz-plugins/issues"
  },
  "homepage": "https://github.com/nielsen-oss/superset-viz-plugins/issues#readme",
  "dependencies": {
    "react": "^16.13.1"
  },
  "babel": {
    "presets": [
      "react-app"
    ]
  },
  "devDependencies": {
    "@emotion/jest": "^11.1.0",
    "@types/recharts": "^1.8.18",
    "jest-styled-components": "^7.0.3",
    "react-bootstrap": "^1.4.0",
    "styled-components": "^5.2.1"
  },
  "peerDependencies": {
    "react-bootstrap": "^1.4.0"
  }
}
