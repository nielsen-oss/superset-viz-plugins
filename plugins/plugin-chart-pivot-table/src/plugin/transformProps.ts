/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { applyDatasourceLabels, extractUniqueData, getOneDimensionData, getUnits, makeDataUnique } from './utils';
import { ShowTotal } from '../types';

type MetricObject<M extends string> = {
  label: M;
  column: {
    verbose_name: M;
  };
};

type FormData<R extends string, C extends string, M extends string> = {
  numberFormat: string;
  numbersFormat: string;
  emptyValuePlaceholder: string;
  compactView: boolean;
  transpose: boolean;
  showTotal: ShowTotal;
  rows: R[];
  columns: C[];
  metrics: MetricObject<M>[];
};

export type QueryData<R extends string, C extends string, M extends string> = Record<R | C, string> & Record<M, number>;
export type DataSource<R extends string, C extends string> = {
  verboseMap: Record<string, R> | Record<string, C>;
};
export type ChartProps<R extends string = string, C extends string = string, M extends string = string> = {
  width: number;
  height: number;
  datasource: DataSource<R, C>;
  formData: FormData<R, C, M>;
  queriesData: {
    data: QueryData<R, C, M>[];
  }[];
};

export default function transformProps<R extends string = string, C extends string = string, M extends string = string>(
  chartProps: ChartProps<R, C, M>,
) {
  const { width, height, formData, queriesData, datasource } = chartProps;
  let { data } = queriesData[0];
  const metrics = formData.metrics.map(({ label }) => label).sort();
  data = applyDatasourceLabels<R, C, M>(data, datasource);
  data = makeDataUnique<R, C, M>(data, metrics);
  const {
    transpose,
    rows: tempRows,
    columns: tempColumns,
    compactView,
    numberFormat,
    numbersFormat: tempNumbersFormat,
    showTotal = ShowTotal.noTotal,
    emptyValuePlaceholder,
  } = formData;
  const numbersFormat = tempNumbersFormat || numberFormat;
  let rows: R[] = (tempRows ?? []).map(row => datasource.verboseMap[row] ?? row) as R[];
  let columns: C[] = (tempColumns ?? []).map(column => datasource.verboseMap[column] ?? column) as C[];
  if (transpose) {
    rows = ((tempColumns as unknown) as R[]) || [];
    columns = ((tempRows as unknown) as C[]) || [];
  }

  const {
    columnUnits,
    rowUnits,
    numberOfColumnsPerMetric,
    numberOfRows,
    oneDimensionRows,
    uiColumnUnits,
    uiRowUnits,
    oneDimensionColumns,
  } = getUnits<R, C, M>(data, columns, rows, metrics);

  const { oneDimensionData, columnsFillData, rowsFillData, rowsTotal, columnsTotal, total } = getOneDimensionData<
    R,
    C,
    M
  >({
    data,
    metrics,
    columns,
    rows,
    columnUnits,
    rowUnits,
    numberOfColumnsPerMetric,
    numbersFormat,
    numberOfRows,
    oneDimensionRows,
    oneDimensionColumns,
    showTotal,
    emptyValuePlaceholder,
  });
  return {
    width,
    height,
    data: oneDimensionData,
    rows,
    emptyValuePlaceholder,
    uiColumnUnits: extractUniqueData(uiColumnUnits),
    uiRowUnits: extractUniqueData(uiRowUnits),
    columnUnits: extractUniqueData(columnUnits),
    rowUnits: extractUniqueData(rowUnits),
    columnsFillData,
    rowsFillData,
    columns,
    total,
    compactView,
    rowsTotal,
    numberOfColumns: numberOfColumnsPerMetric * metrics.length,
    numberOfRows,
    columnsTotal,
    showTotal,
    metrics,
  };
}
