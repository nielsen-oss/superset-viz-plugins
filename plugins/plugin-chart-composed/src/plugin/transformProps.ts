/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { ChartProps } from '@superset-ui/core';
import { AxisInterval } from 'recharts';
import { CHART_SUB_TYPES, CHART_TYPES, Layout, mergeBy } from '../components/utils';
import { ComposedChartProps } from '../components/ComposedChart';
import {
  addBreakdownYColumnsAndGetBreakdownValues,
  addRechartsKeyAndGetXColumnValues,
  checkTimeSeries,
  Data,
  FormData,
  getChartSubType,
  processNumbers,
  QueryMode,
  ResultData,
  SortingType,
  sortOrderedBars,
} from './utils';

export default function transformProps(chartProps: ChartProps) {
  const { width, height, queriesData } = chartProps;
  const data = queriesData[0].data as Data[];
  const formData = chartProps.formData as FormData;

  let xColumns: string[];
  let yColumns: string[];

  if (formData.queryMode === QueryMode.raw) {
    xColumns = [formData.xColumn];
    yColumns = [formData.yColumn];
  } else {
    xColumns = formData.groupby;
    yColumns = formData.metrics?.map(metric => metric.label);
  }

  const xColumnValues: string[] = [];
  const isTimeSeries = checkTimeSeries(xColumns, formData.granularitySqla, formData.layout);
  let resultData: ResultData[] = addRechartsKeyAndGetXColumnValues(
    formData,
    data,
    xColumnValues,
    isTimeSeries,
    xColumns,
  );

  if (isTimeSeries) {
    resultData.sort(
      ({ __timestamp: __timestamp2 }, { __timestamp: __timestamp1 }) =>
        (__timestamp2 as number) - (__timestamp1 as number),
    );
  }

  const breakdowns: string[] = [];
  resultData = addBreakdownYColumnsAndGetBreakdownValues(resultData, yColumns, formData, breakdowns);

  // Unit data elements by groupBy values
  resultData = mergeBy(resultData, 'rechartsDataKey');

  const chartSubType = getChartSubType(
    formData.chartType,
    formData.barChartSubType,
    formData.lineChartSubType,
    formData.areaChartSubType,
    formData.scatterChartSubType,
  );

  const chartTypeMetrics: (keyof typeof CHART_TYPES)[] = [];
  const chartSubTypeMetrics: (keyof typeof CHART_SUB_TYPES)[] = [];
  const useCustomTypeMetrics: boolean[] = [];

  if (formData.queryMode !== QueryMode.raw) {
    yColumns.forEach((yColumn, index) => {
      useCustomTypeMetrics.push(formData[`useCustomTypeMetric${index}`] as boolean);
      chartTypeMetrics.push(formData[`chartTypeMetric${index}`] as keyof typeof CHART_TYPES);
      chartSubTypeMetrics.push(
        getChartSubType(
          formData[`chartTypeMetric${index}`] as keyof typeof CHART_TYPES,
          formData[`barChartSubTypeMetric${index}`] as keyof typeof CHART_SUB_TYPES,
          formData[`lineChartSubTypeMetric${index}`] as keyof typeof CHART_SUB_TYPES,
          formData[`areaChartSubTypeMetric${index}`] as keyof typeof CHART_SUB_TYPES,
          formData[`scatterChartSubTypeMetric${index}`] as keyof typeof CHART_SUB_TYPES,
        ),
      );
    });
  }

  let resultShowTotals = false;
  if (
    formData.barChartSubType === CHART_SUB_TYPES.STACKED &&
    formData.chartType === CHART_TYPES.BAR_CHART &&
    useCustomTypeMetrics.every(el => !el)
  ) {
    resultShowTotals = formData.showTotals;
  }

  const orderByYColumn = formData.useOrderByMetric0 && formData.orderByTypeMetric0;

  const hasOrderedBars = formData.chartType === CHART_TYPES.BAR_CHART && !!orderByYColumn;

  if (hasOrderedBars) {
    sortOrderedBars(
      resultData,
      xColumnValues,
      formData,
      formData.queryMode === QueryMode.raw ? 'XAxisColumn' : 'GroupBy',
    );
  }

  resultData = processNumbers(resultData, breakdowns, formData.numbersFormat, formData.numbersFormatDigits);
  const result: ComposedChartProps = {
    orderByYColumn: orderByYColumn as SortingType,
    hasOrderedBars,
    minBarWidth: formData.minBarWidth,
    breakdowns,
    width,
    height,
    isTimeSeries,
    xColumns,
    yColumns,
    chartTypeMetrics,
    chartSubTypeMetrics,
    hasCustomTypeMetrics: useCustomTypeMetrics,
    layout: formData.layout,
    colorScheme: formData.colorScheme,
    chartType: formData.chartType,
    showLegend: formData.showLegend,
    legendPosition: formData.legendPosition,
    chartSubType,
    showTotals: resultShowTotals,
    numbersFormat: formData.numbersFormat,
    labelsColor: formData.labelsColor,
    xAxis: {
      interval: formData.xAxisInterval as AxisInterval,
      label: formData.xAxisLabel,
      tickLabelAngle: -Number(formData.xAxisTickLabelAngle),
    },
    hasY2Axis:
      yColumns.length > 1 &&
      formData.useY2Axis &&
      formData.layout === Layout.horizontal &&
      formData.queryMode !== QueryMode.raw,
    yAxis: {
      labelAngle: -Number(formData.yAxisLabelAngle ?? 0),
      labelAngle2: -Number(formData.y2AxisLabelAngle ?? 0),
      label: formData.yAxisLabel,
      tickLabelAngle: -Number(formData.yAxisTickLabelAngle),
      label2: formData.y2AxisLabel,
      tickLabelAngle2: -Number(formData.y2AxisTickLabelAngle),
    },
    data: resultData,
  };
  return result;
}
