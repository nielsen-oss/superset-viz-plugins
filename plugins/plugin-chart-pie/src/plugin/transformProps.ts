/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { ChartProps } from '@superset-ui/core';
import { PieChartData, PieProps } from '../PieChart';
import { LegendPosition } from '../utils';

type FormData<G extends string, DK extends string> = {
  colorScheme: string;
  isDonut: boolean;
  groupBy: G;
  metric: {
    label: DK;
  };
  colorPicker: string;
  showLegend: boolean;
  labelType: string;
  showLabels: boolean;
  legendPosition: LegendPosition;
};

export default function transformProps<G extends string, DK extends string>(chartProps: ChartProps): PieProps<G, DK> {
  const { width, height, formData, queryData, queriesData } = chartProps;
  const {
    colorScheme,
    isDonut,
    groupBy,
    metric,
    colorPicker,
    showLegend,
    labelType,
    showLabels,
    legendPosition,
  } = formData as FormData<G, DK>;
  const data = (queriesData?.[0]?.data || queryData.data) as PieChartData<G, DK>[];

  return {
    dataKey: metric.label,
    width,
    legendPosition,
    height,
    data: data.filter(item => item[metric.label] !== null).filter(item => item[groupBy] !== null),
    isDonut,
    baseColor: colorPicker,
    colorScheme,
    showLegend,
    showLabels: isDonut || showLabels,
    groupBy,
    labelType,
  };
}
