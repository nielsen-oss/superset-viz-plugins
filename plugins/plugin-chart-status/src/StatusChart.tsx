/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { SupersetTheme, styled } from '@superset-ui/core';
import { StatusColorsMap } from './types';

interface StatusStylesProps {
  height: number;
  width: number;
}

export type StatusProps = {
  height: number;
  width: number;
  data: [string, string];
  statusColorsMap: StatusColorsMap;
};

const Styles = styled.div<StatusStylesProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: ${({ height }) => height}px;
  width: ${({ width }) => width}px;
`;

const ObjectName = styled.div<{ theme: SupersetTheme }>`
  font-size: ${({ theme }) => theme.typography.sizes.xl}px;
`;

const StatusName = styled.div<{ col: string; theme: SupersetTheme }>`
  font-size: ${({ theme }) => theme.typography.sizes.xxl}px;
  font-weight: ${({ theme }) => theme.typography.weights.bold};
  color: ${({ col }) => col};
`;

export default function StatusChart(props: StatusProps) {
  const { data, height, width, statusColorsMap } = props;
  return (
    <Styles height={height} width={width} data-testid="status">
      <ObjectName>{data[0]}</ObjectName>
      <StatusName col={statusColorsMap[data[1]]}>{data[1]}</StatusName>
    </Styles>
  );
}
