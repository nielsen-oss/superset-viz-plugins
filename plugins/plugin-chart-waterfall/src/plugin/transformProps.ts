/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { ChartProps } from '@superset-ui/core';
import { WaterfallChartData } from '../components/WaterfallChart';
import { convertDataForRecharts, createReChartsBarValues, SortingType } from './utils';
import { LegendPosition } from '../components/utils';

type Metric = {
  label: string;
};

export type QueryData = {
  [key: string]: number | string;
};

type FormData = {
  xAxisColumn: string;
  periodColumn: string;
  queryFields: { metric: string };
  metric: Metric;
  numbersFormat: string;
  legendPosition: LegendPosition;
  orderByChange: SortingType;
  useOrderByChange: boolean;
};

export default function transformProps(
  chartProps: ChartProps,
): {
  dataKey: string;
  onBarClick: () => null;
  numbersFormat: string;
  data: WaterfallChartData[];
  xAxisDataKey: string;
  width: number;
  resetFilters: () => null;
  legendPosition: LegendPosition;
  height: number;
} {
  const { width, height, formData, queriesData } = chartProps;

  const {
    periodColumn,
    xAxisColumn,
    metric,
    numbersFormat,
    legendPosition,
    orderByChange,
    useOrderByChange,
  } = formData as FormData;

  const valueColumn = metric.label;
  const data = queriesData?.[0]?.data as QueryData[];

  const rechartsData = convertDataForRecharts(
    periodColumn,
    xAxisColumn,
    valueColumn,
    data,
    orderByChange,
    useOrderByChange,
  );

  const resultData = createReChartsBarValues(rechartsData, valueColumn, periodColumn);

  return {
    dataKey: valueColumn,
    xAxisDataKey: xAxisColumn,
    width,
    height,
    legendPosition,
    numbersFormat,
    data: resultData,
    onBarClick: () => null,
    resetFilters: () => null,
  };
}
