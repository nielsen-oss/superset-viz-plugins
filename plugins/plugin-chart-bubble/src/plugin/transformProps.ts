/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { ChartProps, DataRecord } from '@superset-ui/core';
import { Metric, SortingType } from '@superset-viz-plugins/plugin-chart-waterfall/src/plugin/utils';
import { LegendPosition } from '@superset-viz-plugins/plugin-chart-waterfall/src/types';

export type QueryData = {
  [key: string]: number | string;
};

type FormData = {
  xAxisColumn: string;
  xAxisLabel: string;
  yAxisLabel: string;
  yAxisLabelAngle: string;
  yAxisLogScale: boolean;
  xAxisTickLabelAngle: string;
  xAxisLogScale: boolean;
  numbersFormat: string;
  colorScheme: string;
  legendPosition: LegendPosition;
  maxNumberOfLegends: string;
  showGridLines: boolean;
  bubbleSize: boolean;
  entity: string;
  series: string;
  xAxis: { label: string } | string;
  yAxis: { label: string } | string;
  zAxis: { label: string } | string;
};

export default function transformProps(chartProps: ChartProps) {
  const { width, height, formData, queriesData } = chartProps;
  const data = queriesData[0].data as QueryData[];

  const {
    xAxisColumn,
    numbersFormat,
    legendPosition,
    maxNumberOfLegends,
    showGridLines,
    bubbleSize,
    xAxisLogScale,
    yAxisLogScale,
    colorScheme,
    entity,
    series,
  } = formData as FormData;

  return {
    xAxisDataKey: xAxisColumn,
    xAxisLabel: formData.xAxisLabel ?? '',
    yAxisLabel: formData.yAxisLabel ?? '',
    xAxisTickLabelAngle: -Number(formData.xAxisTickLabelAngle ?? 45),
    yAxisLabelAngle: -Number(formData.yAxisLabelAngle),
    xAxisLogScale,
    yAxisLogScale,
    width,
    height,
    bubbleSize: bubbleSize ?? 1000,
    legendPosition,
    maxNumberOfLegends,
    showGridLines,
    numbersFormat,
    colorScheme,
    data,
    entity,
    series,
    xAxis: formData.xAxis?.label ?? formData.xAxis,
    yAxis: formData.yAxis?.label ?? formData.yAxis,
    zAxis: formData.zAxis?.label ?? formData.zAxis,
  };
}
