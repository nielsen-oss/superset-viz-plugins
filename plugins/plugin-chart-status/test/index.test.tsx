/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { supersetTheme, ThemeProvider } from '@superset-ui/core';
import StatusChartPlugin from '../src';
import StatusChart from '../src/StatusChart';
import transformProps from '../src/plugin/transformProps';
import { statusesAndObjects } from './__mocks__/statusProps';

jest.mock('recharts');

describe('plugin-chart-status', () => {
  beforeEach(() => {
    // Recharts still have some UNSAFE react functions that failing test
    jest.spyOn(console, 'warn').mockImplementation(() => null);
  });
  const getWrapper = (props: object) =>
    render(
      <ThemeProvider theme={supersetTheme}>
        {/*
       // @ts-ignore (no need emulate all props) */}
        <StatusChart {...transformProps(props)} />
      </ThemeProvider>,
    );
  it('exists', () => {
    expect(StatusChartPlugin).toBeDefined();
  });

  it('Multiple Objects / Statuses', () => {
    getWrapper(statusesAndObjects);
    expect(screen.getByTestId('status')).toMatchSnapshot();
  });
});
