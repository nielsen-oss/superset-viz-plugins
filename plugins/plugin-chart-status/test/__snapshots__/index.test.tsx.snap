// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin-chart-status Multiple Objects / Statuses 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 400px;
  width: 800px;
}

.emotion-1 {
  font-size: 21px;
}

.emotion-2 {
  font-size: 28px;
  font-weight: 700;
  color: rgba(90,193,137,1);
}

<div
  class="emotion-0"
  data-testid="status"
  height="400"
  width="800"
>
  <div
    class="emotion-1"
  >
    2020, 2019, 2017, 2018
  </div>
  <div
    class="emotion-2"
  >
    Healthy, Infected
  </div>
</div>
`;
